tests/certification/bindings/postgres/postgres_test.go

```diff
@@ -0,0 +1,241 @@
+	numOfMessages          = 10
+	dockerConnectionString = "host=localhost user=postgres password=example port=5432 connect_timeout=10 database=dapr_test sslmode=disable"
+)
+
+func TestPostgres(t *testing.T) {
+
+	ports, _ := dapr_testing.GetFreePorts(3)
+	grpcPort := ports[0]
+	httpPort := ports[1]
+
+	testExec := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		require.NoError(t, err, "Could not initialize dapr client.")
+
+		metadata := make(map[string]string)
+
+		ctx.Log("Invoking output binding for exec operation!")
+		req := &daprClient.InvokeBindingRequest{Name: "standard-binding", Operation: "exec", Metadata: metadata}
+		req.Metadata["sql"] = "INSERT INTO dapr_test_table (id, c1, ts) VALUES (1, 'demo', '2020-09-24T11:45:05Z07:00');"
+		errBinding := client.InvokeOutputBinding(ctx, req)
+		require.NoError(ctx, errBinding, "error in output binding - exec")
+
+		return nil
+	}
+
+	testQuery := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		require.NoError(t, err, "Could not initialize dapr client.")
+
+		metadata := make(map[string]string)
+
+		ctx.Log("Invoking output binding for query operation!")
+		req := &daprClient.InvokeBindingRequest{Name: "standard-binding", Operation: "query", Metadata: metadata}
+		req.Metadata["sql"] = "SELECT * FROM dapr_test_table WHERE id = 1;"
+		errBinding := client.InvokeOutputBinding(ctx, req)
+		require.NoError(ctx, errBinding, "error in output binding - query")
+
+		return nil
+	}
+
+	testClose := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		require.NoError(t, err, "Could not initialize dapr client.")
+
+		metadata := make(map[string]string)
+
+		ctx.Log("Invoking output binding for query operation!")
+		req := &daprClient.InvokeBindingRequest{Name: "standard-binding", Operation: "close", Metadata: metadata}
+		errBinding := client.InvokeOutputBinding(ctx, req)
+		require.NoError(ctx, errBinding, "error in output binding - close")
+
+		return nil
+	}
+
+	createTable := func(ctx flow.Context) error {
+		db, err := sql.Open("postgres", dockerConnectionString)
+		assert.NoError(t, err)
+		_, err = db.Exec("CREATE TABLE dapr_test_table(id INT, c1 TEXT, ts TEXT);")
+		assert.NoError(t, err)
+		db.Close()
+		return nil
+	}
+
+	// checks the state store component is not vulnerable to SQL injection
+	verifySQLInjectionTest := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+
+		metadata := make(map[string]string)
+
+		ctx.Log("Invoking output binding for verificatio0n of SQL Injection!")
+		req := &daprClient.InvokeBindingRequest{Name: "standard-binding", Operation: "close", Metadata: metadata}
+		req.Metadata["sql"] = "DROP TABLE dapr_test_table dapr' OR '1'='1"
```

Comment:
What do you think you are testing here?

This isn't a SQL injection test. The binding lets a user run arbitrary SQL, so there is nothing to check.

Also according to your code you are dropping our test table... but then you never check whether the table is still present or not. In a SQL injection test you would want to verify whether any data was modified.

In any case, you should remove the SQL injection test as it simply doesn't apply to this component.